### PR TITLE
Validate that Phone Number is 8 - 16 digits

### LINKED ISSUE #157: [PE-D] Invalid Phone Numbers are allowed
I am able to edit a student to have an invalid phone number `000` using the command `edit 1 p/000`

Although it is unlikely that your target users (the TAs)  would key in such a blatantly invalid phone number, it is entirely possible and human for them
to make minor errors while inputting their student's phone number. *They could miss out 1 digit or key in one extra digit*

Given that your target users would most likely be managing students, perhaps a basic validation check to ensure that the phone number is at least 8 digits long would be appropriate.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: twothicc/ped#5

### PR DESCRIPTION
**Pull request by FergusMok**
Fixes #157, #183

Also, closing #153, as it is redundant. (If the person gives you a wrong email, there's no point in complaining about the format, because it'll bounce either way).

### CODE REVIEW

**Review by uyencfi — APPROVED** [review]
LGTM! 
Although I'm not sure #153 should be closed with this PR because it seems like a different issue. The student might give the wrong email or the user made a typo resulting in an invalid email, but either way it's still an invalid input that we should try to guard against.

File: src/main/java/seedu/edrecord/model/person/Phone.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
     public static final String MESSAGE_CONSTRAINTS =
             "Phone numbers should only contain numbers, and it should be exactly 8 digits long";
```
Comment (uyencfi) [inline src/main/java/seedu/edrecord/model/person/Phone.java:14]: Message should be updated to 8-16 digits too.

**Review by uyencfi — DISMISSED** [review]

File: src/main/java/seedu/edrecord/model/person/Phone.java (hunk 1 above)
Comment (FergusMok) [reply src/main/java/seedu/edrecord/model/person/Phone.java]: Updated, thanks for catching it

**Review by chaiwanlin — APPROVED** [review]
lgtm

**Review by caipng — APPROVED** [review]

**Pull request merged**